drivers: modem: wncm14a2a: remove socket_reading logic

mike-scott opened the pull request:
Remove overly complicated logic to skip incoming data if we were
still waiting on a previous set of data to be read.

This fixes a bug where an error during data receive could end up
with the modem ignoring all incoming data.

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott committed:
drivers: modem: wncm14a2a: remove socket_reading logic

Remove overly complicated logic to skip incoming data if we were
still waiting on a previous set of data to be read.

This fixes a bug where an error during data receive could end up
with the modem ignoring all incoming data.

Signed-off-by: Michael Scott <mike@foundries.io>

pfalcon commented on drivers/modem/wncm14a2a.c:
```diff
@@ -946,23 +942,14 @@ static void on_cmd_sockdataind(struct net_buf **buf, u16_t len)
+			 sock->socket_id, left_bytes);
+
+		/* We still have a lock from hitting this cmd trigger,
+		 * so don't hold one when we send the new command
```
Based on the code, I'd find this message confusing. Did you mean "so don't acquire a new one when ...". Or "so "borrow" it to the func sending new command" ?

mike-scott replied:
(This comment wasn't changed -- the indentation was altered when the surrounding if check was removed.)

However, it's confusing for several reasons:
1) The "lock" is actually a semaphore taken in send_at_cmd() when there is a valid timeout param.  The semaphore is given back in a response handler or it times out.
2) We actually don't have a semaphore from a previous call to send_at_cmd(), this function is triggered via an unsolicited modem trigger "@SOCKDATAIND" which let's us know that there is data ready to be read from the modem.
3) The real reason we don't want to wait for a response here has more to do with the response that the modem generates when we send the command to read the waiting data.  It doesn't send OK or ERROR after echoing the command like most other modem commands, it sends back the data we asked for and THEN an OK/ERROR.   For that reason, there is another command handler on_cmd_sockread() which handles the incoming data asynchronously.  If there's an error, we let the on_cmd_sockerror() function handle it.

**I will add a 2nd commit updating the comment to reflect the above.**

Here's an example of the modem communication flow:
[IN] @SOCKDATAIND:1,0,4 (we have 4 bytes of waiting data)
^^ This is what triggers the on_cmd_sockdataind() handler

[OUT] AT@SOCKREAD=1,4 (we ask for the waiting data .. 4 bytes)

[IN] AT@SOCKREAD=1,4 (modem echoes back any  commands)
^^ This trigger is ignored via a generic on_cmd_atcmdecho() function
[IN] @SOCKREAD:4,"60440024" (4 bytes of ascii-hex encoded data)
^^ Triggers the on_cmd_sockread() handler
[IN] OK
[IN] @SOCKDATAIND:1,0,0 (we have 0 bytes of waiting data -- triggered after every @SOCKREAD call)
^^ This triggers a second call of the on_cmd_sockdataind() handler, but with a 0 length

pfalcon reviewed (approved):
I'd tend to just approve this because it comes from the driver maintainer.

mike-scott committed:
drivers: modem: wncm14a2a: correct comment in on_cmd_sockdataind()

The comment was incorrect explaining why we were sending an
AT-command without waiting for a response (via a K_NO_WAIT timeout).

Let's correct the comment and avoid confusion.

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott commented:
@pfalcon let me know if the comment makes more sense now.

pfalcon commented:
@mike-scott: Yes, it's definitely much more detailed, and will help anyone who will need to debug this driver (or use it as a base for other hardware). Thanks!

mike-scott commented:
@jukkar gentle ping (welcome back from vacation!)
